llvm/lib/Object/DXContainer.cpp

```diff
@@ -228,6 +242,53 @@ void DXContainer::PartIterator::updateIteratorImpl(const uint32_t Offset) {
   IteratorState.Offset = Offset;
 }
 
+Error DirectX::RootSignature::parse(StringRef Data) {
+  const char *Current = Data.begin();
+
+  // Root Signature headers expects 6 integers to be present.
+  if (Data.size() < 6 * sizeof(uint32_t)) {
+    return parseFailed("Invalid data. Too small.");
+  }
```

Comment:
nit: https://llvm.org/docs/CodingStandards.html#don-t-use-braces-on-simple-single-statement-bodies-of-if-else-loop-statements

```suggestion
  if (Data.size() < 6 * sizeof(uint32_t))
    return parseFailed("Invalid data. Too small.");
```

Reply:
Missed that, thanks